### ISSUE TITLE
Incorrect next date when using legacy mode

### ISSUE THREAD

**Issue by undsoft** [Hexagon/croner#64]
```
const cron = new Cron('15 9 * * mon', {
  legacyMode: true,
});
const borked = cron.next(new Date('2022-02-28 23:59:00'));
console.log(borked);
```

Expected: Something in March.
Actual: Mon Feb 28 2022 09:15:00

**Comment by Hexagon** [comment]
Oh, will add more tests for legacyMode, and hopefully a fix, asap. Thanks!

**Comment by Hexagon** [comment]
Resolved in 4.2.2
